Dockerfile-withgo

```diff
@@ -0,0 +1,26 @@
+# This Dockerfile is used for cluster testing - it produces a much larger image
```

Comment:
Is it possible to move this file into `internal/clustertests`?
It would also be nice to include the main Dockerfile dynamically and add to it, but it seems we can't easily do that without something like [this](https://github.com/moby/moby/issues/4027).

Reply:
We need to be able to include the current Pilosa code tree when building (for easy testing of changes), hence the COPY command. `COPY ../..` is not allowed because of:

```
Step 3/13 : COPY ../.. /go/src/github.com/pilosa/pilosa/
COPY failed: Forbidden path outside the build context: ../.. ()
```

I agree it would be very nice to be able to base the build off another Dockerfile. 

I don't see a way to include the current Pilosa codebase without having an extra Dockerfile at the top level though since our current image uses a builder and doesn't include the codebase.